### ISSUE TITLE
Automate issuing of releases

### ISSUE THREAD

**Issue by oleg68** [GrandOrgue/grandorgue#711]
Now the process of issuing of releases is the following

1. Create a release tag and push it to master
2. Wait when the release auto build creates the release on github and update its description manually with the CHANGELOG.md contents

(1) was acceptable when GO was hosted in a private repository. But working with organisation means no direct access to the GO repositories (I disable it in my git settings for prudence) and making all changes in own fork with creating a pull request. But Github does not allow PR on tags. It requires to enable direct access to the target repo temporary.

Moreover issuing releases should have aproval from other members. Tagging does not allow this.

(2) requires manual actions and may be done incorrectly.

My approach of the new process is the following:

1. Add the new release label on top of CHANGELOG.md
2. Make a ususa PR with the only change in this file
3. After approval and merging this PR all futher steps should be made authomatically. 

@larspalo @rousseldenis how do you find the new process?

**Comment by rousseldenis** [comment]
In the processes I'm used to, people that have write access will create releases.

Usually, updating the changelog and using tags. IMHO, the use of tags should be mandatory as they allow to freeze a version. 

Concerning the approval of other members, maybe yes or maybe not, I'm not sure this will help in the flow.

PS: A little tool that helps releases is bumpversion. It allows to modify several files at once with --patch/minor/major commands.

**Comment by rousseldenis** [comment]
> Usually, updating the changelog and using tags. IMHO, the use of tags should be mandatory as they allow to freeze a version

And the main aim too is that allows to get back to that version if master is in a mess.

**Comment by oleg68** [comment]
> IMHO, the use of tags should be mandatory as they allow to freeze a version.

I suggest the build script to read the first line of CHANGELOG.md and to put the tag authomatically.  Not to put the version tag manually.

**Comment by rousseldenis** [comment]
> > IMHO, the use of tags should be mandatory as they allow to freeze a version.
> 
> I suggest the build script to read the first line of CHANGELOG.md and to put the tag authomatically. Not to put the version tag manually.

Of course, if we can make our path in order to get them automatically, let's go

**Comment by larspalo** [comment]
I like the process Oleg suggests.
